File: Data/Models/Team.cs
namespace Blackwood.Access.Models
{
	using System;
	using System.ComponentModel.DataAnnotations;

	public class Team
	{
		// public int ID { get; set;}

		[Key]
		public Int16 TeamCode { get; set; }

		// public Guid CareSysLocationGuid { get; set; }

		public string TeamDesc { get; set; }

		// public bool EnableSync { get; set; }

		// public string PrimaryContact { get; set; }
	}
}

File: ClientApp/app/models/payroll/Team.ts
import { Guid } from "../Utilities";

import { LocalAuthority } from "../reporting/LocalAuthority";
import { Service } from "../reporting/Service";
import { ServiceLocation } from "./ServiceLocation";
import { TeamUser } from "../integration/TeamUser";

export class Team {
	public id: number;
	public teamCode: number;
	public caresysLocationGuid: Guid;
	public teamDesc: string;
	public enableSync: true;
	public primaryContact: string;
	public costCentre: string;
	public breakPolicy: number;
	public locAuthRef: string;
	public hourlyCalc: HourlyCalc;

	public localAuthority: LocalAuthority;
	public serviceLocation: ServiceLocation;

	public teamUsers: TeamUser[];
	
	// Deprecated, to be removed in due course
	public serviceId: number;
	public service: Service;
}

export enum HourlyCalc {
	ContractedAverage = 0,
	ScheduledAvail = 1
}